Content.Server/Chemistry/ReagentEffects/Electrocute.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using Content.Server.Atmos.Components;
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Electrocution;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.Chemistry.ReagentEffects;
+
+public class Electrocute : ReagentEffect
+{
+    [DataField("electrocuteTime")] public int ElectrocuteTime = 2;
+
+    [DataField("electrocuteDamageScale")] public int ElectrocuteDamageScale = 5;
+
+    public override void Effect(ReagentEffectArgs args)
+    {
+        EntitySystem.Get<ElectrocutionSystem>().TryDoElectrocution(args.SolutionEntity, null,
+            Math.Max((args.Quantity * ElectrocuteDamageScale).Int(), 1), TimeSpan.FromSeconds(ElectrocuteTime));
+
+        args.Source?.RemoveReagent(args.Reagent.ID, args.Quantity);
```

Comment:
is this for stuff like reactive effects so it can remove some reagent? should prolly have a bool datafield to disable this

Reply:
i'm not actually sure why it has to remove, I just copied ExtinguishReaction

Reply:
metabolism will remove it automatically. here it seems like the behavior is "remove some extra reagent if it electrocutes you"